Example/WalletApp/PresentationLayer/Wallet/SessionProposal/SessionProposalInteractor.swift

```diff
@@ -11,8 +11,12 @@ final class SessionProposalInteractor {
         
         let supportedRequiredChains = proposal.requiredNamespaces["eip155"]?.chains
         let supportedOptionalChains = proposal.optionalNamespaces?["eip155"]?.chains ?? []
-        let supportedChains = (supportedRequiredChains ?? []).union(supportedOptionalChains) ?? []
-        
+        var supportedChains = (supportedRequiredChains ?? []).union(supportedOptionalChains) ?? []
```

Comment:
Prolly a nitpick but, would be useful to make `supportedChains` a Set in case `supportedRequiredChains` and `supportedOptionalChains` have repeated chains? Like `[eip155:1, eip155:137, eip155:10]` and `[eip155:137, eip155:10, eip155:56, .....]`

Reply:
union is a Set method :D

Reply:
Nice! I've told you I'm rusty :D

Reply:
I think it applies to every language